components/director/internal/systemfetcher/handler.go

```diff
@@ -59,169 +63,146 @@ func (h *handler) ScheduleAggregationForSystemFetcherData(writer http.ResponseWr
-						http.Error(writer, "Check for External Tenant failed", http.StatusInternalServerError)
-						return
-					}
-				}
-				businessTenantMappingID, err = h.getBusinessTenantMappingByExternalTenant(ctx, payload.TenantID)
-				if err != nil {
-					log.C(ctx).WithError(err).Errorf("Getting external tenant with id %q failed", payload.TenantID)
-					http.Error(writer, "Getting External Tenant failed", http.StatusInternalServerError)
+			if !apperrors.IsNotFoundError(err) {
+				log.C(ctx).WithError(err).Errorf("Getting tenant by internal id %q failed", tenantID)
+				return
+			}
+
+			businessTenantMapping, err = h.getBusinessTenantMappingByExternalID(ctx, tenantID)
+			if err != nil {
+				if apperrors.IsNotFoundError(err) {
+					log.C(ctx).WithError(err).Errorf("External tenant with id %q not found", tenantID)
 					return
 				}
-			} else {
-				log.C(ctx).WithError(err).Errorf("Getting tenant with id %q failed", payload.TenantID)
-				http.Error(writer, "Getting Tenant failed", http.StatusInternalServerError)
+				log.C(ctx).WithError(err).Errorf("Getting external tenant with id %q failed", tenantID)
 				return
 			}
 		}
 
-		businessTenantMapping, err := h.getBusinessTenantMappingByID(ctx, businessTenantMappingID)
-		if err != nil || businessTenantMapping == nil {
-			if err != nil {
-				log.C(ctx).WithError(err).Error("Loading Business Tenant Mapping for System Fetcher data aggregation failed")
-			} else {
-				log.C(ctx).Error("Loading Business Tenant Mapping for System Fetcher data aggregation failed")
-			}
-			http.Error(writer, "Loading Business Tenant Mapping for System Fetcher data aggregation failed", http.StatusInternalServerError)
+		if businessTenantMapping == nil {
+			log.C(ctx).Error("Loading Business Tenant Mapping for System Fetcher data aggregation failed")
 			return
 		}
 
 		if businessTenantMapping.Type != tenant.Account && businessTenantMapping.Type != tenant.Customer {
 			log.C(ctx).Infof("Tenant with ID %q is of type %q - operations are created only for tenants of type Account and Customer.", businessTenantMapping.ID, businessTenantMapping.Type)
-			writer.WriteHeader(http.StatusOK)
 			return
 		}
 
-		now := time.Now()
-		data := NewSystemFetcherOperationData(businessTenantMappingID)
-		rawData, err := data.GetData()
-		if err != nil {
-			log.C(ctx).WithError(err).Error("Preparing Operation for System Fetcher data aggregation failed")
-			http.Error(writer, "Preparing Operation for System Fetcher data aggregation failed", http.StatusInternalServerError)
-			return
-		}
-
-		newOperationInput := &model.OperationInput{
-			OpType:    model.OperationTypeSystemFetching,
-			Status:    model.OperationStatusScheduled,
-			Data:      json.RawMessage(rawData),
-			Priority:  int(operationsmanager.HighOperationPriority),
-			CreatedAt: &now,
-		}
-
-		opID, err := h.opMgr.CreateOperation(ctx, newOperationInput)
+		opID, err := h.createSystemFetchingOperation(ctx, businessTenantMapping.ID)
```

Comment:
Implementation of createSystemFetchingOperation always schedule operations that are newly created. 
Probably we should propagate the "skipReschedule" flag to it.

Reply:
Why is it needed? We want newly created operations to always be scheduled.